src/IssueViz.UnitTests/IssueVisualizationControl/IssueVisualizationViewModelTests.cs

```diff
@@ -0,0 +1,544 @@
+            selectionEventsMock.SetupRemove(m => m.SelectedLocationChanged -= (sender, args) => { });
+           
+            testSubject.Dispose();
+
+            selectionEventsMock.VerifyRemove(x=> x.SelectedIssueChanged -= It.IsAny<EventHandler<IssueChangedEventArgs>>(), Times.Once);
+            selectionEventsMock.VerifyRemove(x=> x.SelectedFlowChanged -= It.IsAny<EventHandler<FlowChangedEventArgs>>(), Times.Once);
+            selectionEventsMock.VerifyRemove(x=> x.SelectedLocationChanged -= It.IsAny<EventHandler<LocationChangedEventArgs>>(), Times.Once);
+            selectionEventsMock.Verify(x=> x.Dispose(), Times.Once);
+        }
+
+        #endregion
+
+        private void RaiseIssueChangedEvent(IAnalysisIssueVisualization issue)
+        {
+            selectionEventsMock.Raise(x => x.SelectedIssueChanged += null, new IssueChangedEventArgs(issue));
+        }
+
+        private void RaiseFlowChangedEvent(IAnalysisIssueFlowVisualization flow)
+        {
+            selectionEventsMock.Raise(x => x.SelectedFlowChanged += null, new FlowChangedEventArgs(flow));
+        }
+
+        private void RaiseLocationChangedEvent(IAnalysisIssueLocationVisualization location)
+        {
+            selectionEventsMock.Raise(x => x.SelectedLocationChanged += null, new LocationChangedEventArgs(location));
+        }
+
+        private void VerifyNotifyPropertyChanged(params string[] changedProperties)
+        {
+            foreach (var changedProperty in changedProperties)
+            {
+                propertyChangedEventHandler.Verify(x =>
+                        x(It.IsAny<object>(), It.Is((PropertyChangedEventArgs e) => e.PropertyName == changedProperty)),
+                    Times.Once);
+            }
+
+            propertyChangedEventHandler.VerifyNoOtherCalls();
+        }
+
+        private IAnalysisIssueLocationVisualization CreateMockLocation(string filePath, object imageMoniker, Exception failsToRetrieveMoniker = null)
+        {
+            if (failsToRetrieveMoniker != null)
+            {
+                imageServiceMock
+                    .Setup(x => x.GetImageMonikerForFile(filePath))
+                    .Throws(failsToRetrieveMoniker);
+            }
+            else
+            {
+                imageServiceMock.Setup(x => x.GetImageMonikerForFile(filePath)).Returns((ImageMoniker) imageMoniker);
+            }
+
+            var location = new Mock<IAnalysisIssueLocation>();
+            location.Setup(x => x.FilePath).Returns(filePath);
+
+            var locationViz = new Mock<IAnalysisIssueLocationVisualization>();
+            locationViz.Setup(x => x.Location).Returns(location.Object);
+
+            return locationViz.Object;
+        }
+
+        private void VerifyLocationList(IEnumerable<ILocationListItem> expectedLocationsList)
+        {
+            testSubject.LocationListItems.Should().BeEquivalentTo(expectedLocationsList, assertionOptions =>
+                assertionOptions
```

Comment:
1. RespectingRuntimeTypes = Since the list is ILocationListItem, this allows to compare each item in the list based on the expected actual implementation. Otherwise, it will attempt to compare only the properties that exist on ILocationListItem.
2. ComparingByMembers<ImageMoniker> = ImageMoniker is a struct, so this is a way to compare by fields and properties of the struct.

Reply:
You could add that info to relevant lines:
```
  .RespectingRuntimeTypes() // check underlying types rather than `ILocationListItem`
  .ComparingByMembers<ImageMoniker>()); // check struct fields and properties
```

If it's sufficiently obscure to call out in a review comment then it's probably worth capturing in the code to help whoever is maintaining the code in the future.